bzzeth/bzzeth.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package bzzeth
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/network"
+	"github.com/ethersphere/swarm/p2p/protocols"
+	"github.com/ethersphere/swarm/storage"
+)
+
+// BzzEth implements node.Service
+var _ node.Service = &BzzEth{}
+
+// BzzEth is a global module handling ethereum state on swarm
+type BzzEth struct {
+	peers    *peers            // bzzeth peer pool
```

Comment:
why not just directly use `map[enode.ID]*Peer`? This is the convention we use in the rest of the codebase

Reply:
cleaner code like this

Reply:
peers struct actually maintains a map with lock like this..